launcher/ui/pages/instance/ModFolderPage.cpp

```diff
@@ -34,368 +34,95 @@
-        ) {
-            return QSortFilterProxyModel::lessThan(source_left, source_right);
-        }
-
-        // we are now guaranteed to have two valid indexes in the same column... we love the provided invariants unconditionally and proceed.
-
-        auto column = (ModFolderModel::Columns) source_left.column();
-        bool invert = false;
-        switch(column) {
-            // GH-2550 - sort by enabled/disabled
-            case ModFolderModel::ActiveColumn: {
-                auto dataL = source_left.data(Qt::CheckStateRole).toBool();
-                auto dataR = source_right.data(Qt::CheckStateRole).toBool();
-                if(dataL != dataR) {
-                    return dataL > dataR;
-                }
-                // fallthrough
-                invert = sortOrder() == Qt::DescendingOrder;
-            }
-            // GH-2722 - sort mod names in a way that discards "The" prefixes
-            case ModFolderModel::NameColumn: {
-                auto dataL = model->data(model->index(source_left.row(), ModFolderModel::NameColumn)).toString();
-                RemoveThePrefix(dataL);
-                auto dataR = model->data(model->index(source_right.row(), ModFolderModel::NameColumn)).toString();
-                RemoveThePrefix(dataR);
-
-                auto less = dataL.compare(dataR, sortCaseSensitivity());
-                if(less != 0) {
-                    return invert ? (less > 0) : (less < 0);
-                }
-                // fallthrough
-                invert = sortOrder() == Qt::DescendingOrder;
-            }
-            // GH-2762 - sort versions by parsing them as versions
-            case ModFolderModel::VersionColumn: {
-                auto dataL = Version(model->data(model->index(source_left.row(), ModFolderModel::VersionColumn)).toString());
-                auto dataR = Version(model->data(model->index(source_right.row(), ModFolderModel::VersionColumn)).toString());
-                return invert ? (dataL > dataR) : (dataL < dataR);
-            }
-            default: {
-                return QSortFilterProxyModel::lessThan(source_left, source_right);
-            }
-        }
-    }
-};
-
-ModFolderPage::ModFolderPage(
-    BaseInstance *inst,
-    std::shared_ptr<ModFolderModel> mods,
-    QString id,
-    QString iconName,
-    QString displayName,
-    QString helpPage,
-    QWidget *parent
-) :
-    QMainWindow(parent),
-    ui(new Ui::ModFolderPage)
+ModFolderPage::ModFolderPage(BaseInstance* inst, std::shared_ptr<ModFolderModel> mods, QWidget* parent)
+    : ExternalResourcesPage(inst, mods, parent)
 {
-    ui->setupUi(this);
-
     // This is structured like that so that these changes
-    // do not affect the Resouce pack and Shader pack tabs
-    if(id == "mods") {
+    // do not affect the Resource pack and Shader pack tabs
+    {
         auto act = new QAction(tr("Download mods"), this);
         act->setToolTip(tr("Download mods from online mod platforms"));
```

Comment:
Maybe move this to the .ui as well and just hide it for the other pages?

Reply:
I tried doing this in #221, it didn't work because, even if we hide it, there's a space left in the button's place, which looks ugly

Reply:
It isn't possible they said! xD

Reply:
Well, i never said it wasn't possible :p
Besides, what's the point of having it in the ExternalResources UI if only the mods page uses it? And it requires you to change every piece of text / manually insert it either way o.O

Reply:
The idea would be that once Modrinth supports them, we can just reuse that same action on the other pages

Reply:
(also this lets Qt deal with retranslating, as we would otherwise need to do it manually for that action in the `retranslate` function

Reply:
mhm for me it's quite weird only having an action in a ui, without actually putting it in the ui itself, but if it works it works /shrug